packages/core/src/actions/invoke.ts

```diff
@@ -44,29 +43,23 @@ export function invoke<
           }
         } as InvokeActionObject;
       } else {
-        const behaviorImpl = state.machine.options.actors[src.type];
+        const behaviorImpl = state.machine.options.actors[src];
 
         if (!behaviorImpl) {
           resolvedInvokeAction = {
             type,
             params: invokeDef
           } as InvokeActionObject;
         } else {
-          const behavior =
-            typeof behaviorImpl === 'function'
-              ? behaviorImpl(state.context, _event.data, {
-                  id,
-                  data: data && mapContext(data, state.context, _event as any),
-                  src,
-                  _event,
-                  meta
-                })
-              : behaviorImpl;
+          const behavior = behaviorImpl;
 
           const ref = interpret(behavior, {
             id,
             src,
-            parent: actorContext?.self
+            parent: actorContext?.self,
+            input: input
+              ? mapContext(input, state.context, _event as any) // TODO: fix type
```

Comment:
The `mapContext` is typed incorrectly right now and could have a better name. Previously it was only used to map `Partial<LazyValues<TContext, TContext>>` to `Partial<TContext>` (or smth like that), and here it can map `LazyValues<TInput, TContext>` to `TInput`

Reply:
I'm open to renaming this and fixing the types.

Reply:
We probably won't find a good specific name, so let's just use a vague one:
```ts
type ResolvableObject<
  TObject extends {},
  TContext extends MachineContext,
  TEvent extends EventObject
> = TObject | ((context: TContext, event: TEvent) => TObject);

type ResolvableProperties<
  TObject extends {},
  TContext extends MachineContext,
  TEvent extends EventObject
> = {
  [K in keyof TObject]:
    | TObject[K]
    | ((context: TContext, event: TEvent) => TObject[K]);
};

declare function resolveObject<
  TObject extends {},
  TContext extends MachineContext,
  TEvent extends EventObject
>(
  obj:
    | ResolvableObject<TObject, TContext, TEvent>
    | ResolvableProperties<TObject, TContext, TEvent>,
  context: TContext,
  event: TEvent
): TObject;
```

This is very close to the existing definition. It's just that:
1. `resolveObject` accept more generic params - as the argument used for resolving (`TContext`) isn't necessarily the same as the returned value (`TObject`)
2. `ResolvableObject`/`ResolvableProperties` don't contain the optional modifier within the mapped type. The optionality should be provided by the "caller" (`assign` should provide that optionality as it handles "slices" of `TContext`). It shouldn't be assumed by that inner mapped type. After all, when we need to resolve the input - we need all of it, not its subset.

This was just a quick experiment that turned out to be very, very similar to the existing functions (just with different names and very minor tweaks). Feel free to adjust this to your liking, however, you see fit.